core/src/main/scala-3.0.0-RC3/cats/derived/partialOrder.scala

```diff
@@ -0,0 +1,37 @@
+package cats.derived
+
+import cats.PartialOrder
+import shapeless3.deriving.{K0, Complete}
+
+object partialOrder extends PartialOrderDerivation
+
+trait ProductPartialOrder[T[x] <: PartialOrder[x], A](using inst: K0.ProductInstances[T, A])
+    extends PartialOrder[A]:
+
+  def partialCompare(x: A, y: A): Double =
+    inst.foldLeft2(x, y)(0: Double)(
+      [t] => (acc: Double, ord: T[t], t0: t, t1: t) => {
+        val cmp = ord.partialCompare(t0, t1)
+        Complete(cmp != 0)(cmp)(acc)
+      }
+    )
+
+trait CoproductPartialOrder[T[x] <: PartialOrder[x], A](using inst: K0.CoproductInstances[T, A])
+    extends PartialOrder[A]:
+
+  def partialCompare(x: A, y: A): Double =
+    //TODO is this correct? I _think_ it must be to be consistent with Order
```

Comment:
Hmm in Kittens 2 we have the `PartialOrder` where two different subclasses are considered not comparable. I never thought about type class coherence in this case. But OTH it's way more intuitive to me that we can't compare them 🤔

Reply:
Yeah, I guess coherence only really demands that `partialCompare(x,y) != Nan ==> compare(x,y).toDouble == partialCompare(x,y)`. Or at least I would hope that that is the case.

However I can't think of a particularly good reason why we shouldn't compare ordinals when that is considered valid for deriving `Order` instances? Happy to trust your judgement on it though!

Reply:
To me it's the other way around - why should we assume that we can compare two different branches of an ADT? We can make it work for `Order` via `ordinal` but it's kinda arbitrary.

Reply:
Yep, fair enough 👍 Gives another reason to derive `PartialOrder` as well - if you want that behaviour instead of the arbitrary comparison of ordinals